File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>



int count_letters(string text)
{
    int letters = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++; // Counts letter
        }
    }
    return letters;
}

int count_words(string text)
{
    int spaces = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (' ' == text[i])
        {
            spaces++; // Counts spaces
        }
    }

    return spaces + 1;
};

int count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++; // Count sentences
        }
    }

    return sentences;
}

int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    float L = ((float)letters / (float)words) * 100; // The L in the formula
    float S = ((float)sentences / (float)words) * 100; // The S in the formula

    double g = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(g);

    // Prints the grade
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }

}

File: sentimental-readability/readability.py
# TODO

alphabet = "abcdefghijklmnopqrstuvwxyz"
alphabet2 = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"


def count_letters(text):
    letters = 0     # Number of letters
    for i in range(len(text)):
        if text[i] in alphabet or text[i] in alphabet2:     # Check if is a letter
            letters += 1
    return letters


def count_words(text):
    spaces = 0          # Number of spaces

    for i in range(len(text)):
        if text[i] == " ":          # Check if it is a space
            spaces += 1

    return spaces + 1


def count_sentences(text):
    sentences = 0           # Number of sentences

    for i in range(len(text)):          # Check numbers of sentences
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            sentences += 1

    return sentences


if __name__ == "__main__":

    text = input("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    l = (letters / words) * 100     # L in the formula
    s = (sentences / words) * 100   # S in the formula

    g = 0.0588 * l - 0.296 * s - 15.8       se# Formula
    index = round(g)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print("Grade", index)